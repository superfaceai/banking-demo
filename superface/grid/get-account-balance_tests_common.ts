import "dotenv/config";
import { SuperfaceClient } from "@superfaceai/one-sdk";
import { SuperfaceTest } from "@superfaceai/testing";

export function getAccountBalanceTest(provider: string) {
  describe(`chat/channels/${provider}`, () => {
    let superface: SuperfaceTest;

    describe("GetAccountBalance", () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: "banking/get-account-balance",
          provider,
          useCase: "GetAccountBalance",
          testInstance: expect,
        });
      });

      it("performs correctly", async () => {
        await expect(
          superface.run({
            input: undefined,
          })
        ).resolves.toMatchSnapshot();
      });
    });
  });
}
