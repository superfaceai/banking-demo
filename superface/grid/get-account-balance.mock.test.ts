import { getAccountBalanceTest } from "./get-account-balance_tests_common";

getAccountBalanceTest('mock');