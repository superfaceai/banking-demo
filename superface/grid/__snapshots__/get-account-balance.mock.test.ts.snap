// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/channels/mock GetAccountBalance performs correctly 1`] = `
Ok {
  "value": Object {
    "accounts": Array [
      Object {
        "accountType": "CURRENT_ACCOUNT",
        "bic": "DEUTDEFFXXX",
        "currencyCode": "EUR",
        "currentBalance": 100.95,
        "description": "Pers√∂nliches Konto",
        "iban": "DE10010000000000009318",
      },
    ],
  },
}
`;
